File: src/tools/echo.ts
import { McpServer } from '@modelcontextprotocol/sdk/server/mcp.js';
import { z } from 'zod';

export function registerEchoTool(server: McpServer) {
	server.registerTool(
		'echo',
		{
			title: 'Echo',
			description: 'Returns the input message back to the caller',
			inputSchema: z.object({
				message: z.string().describe('The message to echo back'),
			}),
		},
		async ({ message }) => ({
			content: [{ type: 'text', text: message }],
		}),
	);
}

File: openhive/tools/echo.py

from typing import Annotated

from pydantic import Field


def register(mcp):
    @mcp.tool()
    async def echo(
        message: Annotated[str, Field(description="The message to echo back")],
    ) -> str:
        """Echo the input message back to the caller."""
        return message
